pkg/controller/namespace/namespace_controller.go

```diff
@@ -0,0 +1,165 @@
+	// Create a new controller
+	c, err := controller.New("namespace-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Namespace
+	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileNamespace implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileNamespace{}
+
+// ReconcileNamespace reconciles a Namespace object
+type ReconcileNamespace struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a Namespace object and makes changes based on the state read
+// and what is in the Namespace.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Namespace")
+
+	// Fetch the Namespace instance
+	instance := &corev1.Namespace{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	namespaceList := &corev1.NamespaceList{}
+	opts := client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, namespaceList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get namespaceList")
+		return reconcile.Result{}, err
+	}
+
+	subjectPermissionList := &managedv1alpha1.SubjectPermissionList{}
+	opts = client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, subjectPermissionList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get clusterRoleBindingList")
+		return reconcile.Result{}, err
+	}
+
+	clusterRoleList := &v1.ClusterRoleList{}
+	opts = client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, clusterRoleList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get clusterRoleList")
+		return reconcile.Result{}, err
+	}
+
+	// loop through all subject permissions
```

Comment:
Remember to add in the "why." The `for` loop lets us know we're looping through the subject permissions, but not the "why."

A comment at the top of this most outer loop will give the reader a good indication as to what this next 35 lines of code are up to.